MapleServer2/Tools/ConditionHelper.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace MapleServer2.Tools;
+
+public static class ConditionHelper
+{
+    /// <summary>
+    /// Checks if the given condition code or target match the given string value.
+    /// </summary>
+    /// <param name="conditionType">Condition code or target</param>
+    /// <param name="value">Value to try match</param>
+    public static bool IsMatching(string conditionType, string value)
+    {
+        return conditionType.Equals(value, StringComparison.OrdinalIgnoreCase);
+    }
+
+    /// <summary>
+    /// Checks if the given condition codes or targets match the given long value.
+    /// </summary>
+    /// <param name="conditionType">Conditions codes or targets</param>
+    /// <param name="value">Value to try match</param>
+    public static bool IsMatching(string conditionType, long value)
+    {
+        // Conditions can have ranges and lists, so we need to check both
+        if (conditionType.Contains('-') && conditionType.Contains(','))
+        {
+            string[] conditionsValues = conditionType.Split(',');
+            foreach (string conditionValue in conditionsValues)
+            {
+                if (conditionValue.Contains('-') && IsInRange(conditionValue, value))
+                {
+                    return true;
+                }
+
+                if (IsMatching(conditionValue, value))
+                {
+                    return true;
+                }
+            }
+        }
+
+        // Check if the value is in the range
+        if (conditionType.Contains('-') && IsInRange(conditionType, value))
+        {
+            return true;
+        }
+
+        // Check if the value is in the list
+        if (conditionType.Contains(',') && IsInList(conditionType, value))
+        {
+            return true;
+        }
+
+        return long.TryParse(conditionType, out long parsedCondition) && parsedCondition == value;
+    }
+
+    public static bool IsInList(string conditionList, long value)
```

Comment:
Do these need to be public? I think we could probably get away with only `IsMatching` being public?

Reply:
just fixed

Reply:
Wait sorry, I was thinking in making it private but decided it could be useful in the future.

Reply:
I don't think those methods will ever be useful publicly. In order to figure out which one to use, you'll have to duplicate the logic of for example seeing if there's a `-` in them. I think by making those methods public, we'll be encouraging people to duplicate that logic.

Reply:
We never know that the company has in stock for us but sure i'll make them private